src/App.jsx

```diff
@@ -1,38 +1,33 @@
-import React, {useState} from 'react';
+import {useState, useEffect} from 'react';
 import './App.css';
 import TodoList from './components/TodoList';
 import AddTodoForm from './components/AddTodoForm';
 
 
 const Heading = () => <h1>Todo list</h1>
 
-const App = () => {
-  // const todoList = [
-  //   {
-  //     id: 1, 
-  //     title: "Complete assignment"
-  //   },
-  //   {
-  //     id: 2, 
-  //     title: "Read the book"
-  //   },
-  //   {
-  //     id: 3, 
-  //     title: "Watch the video class"
-  //   },
-  // ];
-  // creating state
+const useSemiPersistentState = (key, initialValue) => {
+  const [value, setValue] = useState(
+    JSON.parse(localStorage.getItem(key)) || initialValue
+  );
   
-  const [todoList, setTodoList] = useState([]);
+    useEffect(() => {
+    localStorage.setItem(key, JSON.stringify(value));
+  }, [key, value]);
+  return [value, setValue];
+};
+
+const App = () => {
+  const [todoList, setTodoList] = useSemiPersistentState("savedTodoList", "");
```

Comment:
the default value passed, the empty string, seems inconsistent in terms of typing, todoList is expected to be an array of objects. so I'd suggest making the default value an empty array rather than an empty string